modules/Emails/EmailUI.php

```diff
@@ -422,6 +422,24 @@ public function generateComposePackageForQuickCreateFromComposeUrl($emailLinkUrl
         return $this->generateComposePackageForQuickCreate($a_composeData, $emailLinkUrl, $lazyLoad);
     }
 
+    private function createEmailLink($module_name, $record_id, $name, $addr, $text) {
+        global $current_user;
```

Comment:
Using globals is never the right way. Pass it as an optional dependecy parameter and use it only if it's necessary.
(btw I like the way how you solve a tons of code duplication here :+1: )

Reply:
Hm, I can only move it to the caller, use global there and then pass it 5 times, which seems worse to me. Any thoughts on this?

Reply:
Legacy code using global variables, I am just trying to avoid them as many places I can, doing this in a way like this:
```
public function doStuff(User $currentUser) {
  ...
}
```
(also I working in a wrapper class over the globals and superglobals - unfortunatelly not finished yet)

Of course, if it's not possible to avoid a global variable in a point I will approve it, I just would ask you to help me to avoid these global usages instead adding new if it's possible.
Btw, you'r right, ignore my previous comment.
Thank you

Reply:
> Of course, if it's not possible to avoid a global variable in a point I will approve it, I just would ask you to help me to avoid these global usages instead adding new if it's possible.

Sounds good, I'll try to keep that in mind in the future :+1: